pkg/karmadactl/join.go

```diff
@@ -285,7 +284,7 @@ func registerClusterInControllerPlane(opts CommandJoinOption, controlPlaneRestCo
 		return fmt.Errorf("failed to create impersonator secret in control plane. error: %v", err)
 	}
 
-	cluster, err := generateClusterInControllerPlane(controlPlaneRestConfig, clusterConfig, opts, *secret)
+	cluster, err := generateClusterInControllerPlane(controlPlaneRestConfig, clusterConfig, opts, *secret, *impersonationSecret)
```

Comment:
After generates the `Cluster`, we should patch the `ownerReference` to the secret, just like what we did on the previous `secret`.

Reply:
Ignore it as it has been done by the previous PR.